packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -427,6 +452,144 @@ void _cleanUpBitmapConversionCaches() {
   _bitmapBlobUrlCache.clear();
 }
 
+/// Converts a [BitmapDescriptor] into a [Node] that can be used as
+/// [AdvancedMarker]'s icon.
+Future<web.Node?> _advancedMarkerIconFromBitmapDescriptor(
+  BitmapDescriptor bitmapDescriptor, {
+  required double? opacity,
+  required bool isVisible,
+  required double? rotation,
+}) async {
+  if (bitmapDescriptor is PinConfig) {
+    final gmaps.PinElementOptions options =
+        gmaps.PinElementOptions()
+          ..background =
+              bitmapDescriptor.backgroundColor != null
+                  ? _getCssColor(bitmapDescriptor.backgroundColor!)
+                  : null
+          ..borderColor =
+              bitmapDescriptor.borderColor != null
+                  ? _getCssColor(bitmapDescriptor.borderColor!)
+                  : null;
+
+    final AdvancedMarkerGlyph? glyph = bitmapDescriptor.glyph;
+    switch (glyph) {
+      case final CircleGlyph circleGlyph:
+        options.glyphColor = _getCssColor(circleGlyph.color);
+      case final TextGlyph textGlyph:
+        final web.HTMLParagraphElement element = web.HTMLParagraphElement();
+        element.text = textGlyph.text;
+        if (textGlyph.textColor != null) {
+          element.style.color = _getCssColor(textGlyph.textColor!);
+        }
+        options.glyph = element;
+      case final BitmapGlyph bitmapGlyph:
+        final web.Node?
+        glyphBitmap = await _advancedMarkerIconFromBitmapDescriptor(
+          bitmapGlyph.bitmap,
+          // Always opaque, opacity is handled by the parent marker.
+          opacity: 1.0,
+          // Always visible, as the visibility is handled by the parent marker.
+          isVisible: true,
+          rotation: rotation,
+        );
+        options.glyph = glyphBitmap;
+      case null:
+        break;
+    }
+
+    final gmaps.PinElement pinElement = gmaps.PinElement(options);
+    final web.HTMLElement htmlElement = pinElement.element;
+    htmlElement.style
+      ..visibility = isVisible ? 'visible' : 'hidden'
+      ..opacity = opacity?.toString() ?? '1.0'
+      ..transform = rotation != null ? 'rotate(${rotation}deg)' : '';
+    return htmlElement;
```

Comment:
These implementations are all quite long, which makes it hard to see the overall structure of the method being essentially a switch on `bitmapDescriptor`. This would be much easier to follow if each of the bodies were extracted to its own helper, and this method were just the branching.